keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -472,6 +484,21 @@ public DoubleTensor logInPlace() {
         return this;
     }
 
+    /**
+     * This is identical to log().times(y), except that it changes NaN results to 0.
+     * This is important when calculating 0log0, which should return 0
+     * See https://arcsecond.wordpress.com/2009/03/19/0log0-0-for-real/ for some mathematical justification
+     * @param y The tensor value to multiply by
+     * @return the log of this tensor multiplied by y
+     */
+    @Override
+    public DoubleTensor safeLogTimesInPlace(DoubleTensor y) {
+        TensorValidator.NAN_CATCHER.validate(this);
```

Comment:
Isn't it a bit aggressive to throw an exception if there is NaN in the tensor?

Reply:
Without thinking too much about it, I'd expect values that at some point evaluate to `NaN` to stay `NaN`, so that the operations are not interrupted for the rest of the values in the tensor?

Reply:
👍

Reply:
I could implement it so if there's any `NaN`s in the tensors before the operation, they get put back into the result afterwards. It wouldn't be too hard to do, but I question if that's what we want. Happy to put the question up for debate.

Reply:
After discussion with the team - I've now implemented this as debug-only. In other words, by default the `NAN_CATCHER` and `NAN_FIXER` do nothing, unless you enable them programatically.